cost/aws/old_snapshots/aws_delete_old_snapshots.pt

```diff
@@ -720,19 +735,33 @@ define delete_snapshot($item) do
       "SnapshotId.1": strip($item["id"])
     }
   )
+  call check_response_and_append_if_error($delete_response, "Delete Snapshot", $errors) retrieve $errors
   $splitResult = $delete_response["code"]
   if $splitResult != 200
    call sys_log("Inside delete_snapshot definition", to_s($delete_response))
   end
 end
 
+define skip_error_and_append($subject) do
+  $$errors << "Unexpected error for " + $subject + "\n  " + to_s($_error)
+  $_error_behavior = "skip"
+end
+
+define check_response_and_append_if_error($response, $request_description, $errors) return $errors do
+  if $response["code"] > 299 || $response["code"] < 200
+    $errors << "Unexpected status code from " + $request_description + " request\n  " + to_s($response)
+  end
+end
+
 define sys_log($subject, $detail) do
-  rs_cm.audit_entries.create(
-    notify: "None",
-    audit_entry: {
-      auditee_href: @@account,
-      summary: "AWS Old Snapshots :- " + $subject,
-      detail: $detail
-    }
-  )
+  if $$debug
+    rs_cm.audit_entries.create(
+      notify: "None",
+      audit_entry: {
+        auditee_href: @@account,
+        summary: "AWS Old Snapshots Policy :- " + $subject,
+        detail: $detail
+      }
+    )
+  end
```

Comment:
As noted in Teams, we should consider automatically handling possible situations where creating an audit entry would fail (i.e. shard, permissions).

Reply:
I have now implemented that by skipping and appending any errors from the `sys_log` definition to the `$$errors` global.

Reply:
Rock on!